src/test/java/net/datafaker/idnumbers/EsMXIdNumberTest.java

```diff
@@ -1,29 +1,31 @@
 package net.datafaker.idnumbers;
 
 import net.datafaker.providers.base.BaseFaker;
+import net.datafaker.providers.base.IdNumber;
 import org.junit.jupiter.api.RepeatedTest;
 
 import java.util.Locale;
+import java.util.regex.Pattern;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 class EsMXIdNumberTest {
 
+    private static final Pattern REGEX = Pattern.compile("[A-Z][A-Z][A-Z][A-Z]\\d{6}[HM]" +
```

Comment:
Why not simplify [A-Z]{4} and then 5 for the second set?

Reply:
yes, that makes sense, thanks